pkg/prove/proof.go

```diff
@@ -203,3 +206,105 @@ func splitIntoRows(squareSize uint64, s []shares.Share) [][]shares.Share {
 	}
 	return rows
 }
+
+// SharesInclusion generates an nmt inclusion proof for a set of shares to the data root.
+// expects the shares range to be pre-validated.
+// Note: only supports inclusion proofs for shares belonging to the same namespace.
+func SharesInclusion(
+	codec rsmt2d.Codec,
+	allRawShares []shares.Share,
+	squareSize uint64,
+	namespaceID namespace.ID,
+	startShare uint64,
+	endShare uint64,
+) (types.SharesProof, error) {
+	startRow := startShare / squareSize
+	endRow := endShare / squareSize
+	startLeaf := startShare % squareSize
+	endLeaf := endShare % squareSize
+
+	eds, err := rsmt2d.ComputeExtendedDataSquare(
+		shares.ToBytes(allRawShares),
+		appconsts.DefaultCodec(),
+		wrapper.NewConstructor(squareSize),
+	)
+	if err != nil {
+		return types.SharesProof{}, err
+	}
```

Comment:
This definitely works for now, but if we wanted to dramatically increase the efficiency of this function, in the future we could assume that the user already has, or has access to, the data availability header. If that's the case, then we don't actually have to create the inclusion proofs to the DAH row roots, that can be done by them and we can avoid calculating the entire DAH

Reply:
100% https://github.com/celestiaorg/celestia-app/issues/1050